qemu/tests/boot_order_check.py

```diff
@@ -73,17 +74,26 @@ def run(test, params, env):
     boot_fail_infos = boot_fail_infos % (list_nic_addr[0][0],
                                          list_nic_addr[1][0],
                                          list_nic_addr[2][0])
-
-    error.context("Check the guest boot result", logging.info)
     if bootorder_type == "type2":
-        session_serial = vm.wait_for_serial_login(timeout=timeout)
-        output = vm.serial_console.get_stripped_output()
-        session_serial.close()
+        boot_grap_str = "Booting from Hard Disk.*"
     else:
-        output = vm.serial_console.get_stripped_output()
-        utils_misc.wait_for(lambda: re.search("No bootable device.", output),
-                            timeout, 1)
-        output = vm.serial_console.get_stripped_output()
+        boot_grap_str = "No bootable device."
+
+    error_context.context("Check the guest boot result", logging.info)
+    match = False
+    start = time.time()
+    while True:
+        try:
+            output = vm.serial_console.get_stripped_output()
+        except ValueError:
+            output = vm.serial_console.get_output()
+        match = re.search(boot_grap_str, output)
```

Comment:
In this case, `match = (boot_chk_str in output)` will work just fine.

Reply:
Thanks, updated it.